script/validate_json.rb

```diff
@@ -1,53 +1,87 @@
 #!/usr/bin/env ruby
 
 # Validates JSON files in the _data directory
-# Exits 0 on success
-# Exits 1 upon JSON parsing errors
-# Exits 2 if a file's keys are not in alphanumeric order
-# Exits 3 if any sites.json entries are missing the required 'url' key
+# See ErrorCodes for actual return value
+# Exits 'SUCCESS' on success
+# Exits 'PARSE_FAILED' upon JSON parsing errors
+# Exits 'UNSORTED' if a file's keys are not in alphanumeric order
+# Exits 'MISSING_URL' if any sites.json entries are missing the required 'url' key
+# Exits 'MISSING_DIFFICULTY' if any sites.json entries are missing the required 'difficulty' key
+# Exits 'MISSING_DOMAINS' if any sites.json entries are missing the required 'domains' key
+# Exits 'MISSING_NAME' if any sites.json entries are missing the required 'name' key
+# Exits 'UNEXPECTED_DIFFICULTY' if field contains unexpected value on 'difficulty' key
 
 require 'json'
 
+module ErrorCodes
+    SUCCESS = 0
+    PARSE_FAILED = 1
+    UNSORTED = 2
+    MISSING_URL = 3
+    MISSING_DIFFICULTY = 4
+    MISSING_DOMAINS = 5
+    MISSING_NAME = 6
+    UNEXPECTED_DIFFICULTY = 7
+end
+
 def get_transformed_name(site_object)
     return site_object['name'].downcase.sub(/^the\s+/, '')
 end
 
+def error_on_missing_field(name, key, field, exit_code)
```

Comment:
`name` always references `key['name']`. Remove it from the signature and reference the name field of key in the log message instead
```suggestion
def error_on_missing_field(key, field, exit_code)
```